installinator/src/reporter.rs

```diff
@@ -157,3 +170,135 @@ where
         self.on_tick_task = Some(self.spawn_on_tick_task());
     }
 }
+
+#[derive(Clone, Debug)]
+pub(crate) struct ReportProgressBackend {
+    log: slog::Logger,
+    imp: Arc<dyn ReportProgressImpl>,
+}
+
+impl ReportProgressBackend {
+    pub(crate) fn new<P: ReportProgressImpl + 'static>(
+        log: &slog::Logger,
+        imp: P,
+    ) -> Self {
+        let log = log.new(slog::o!("component" => "ReportProgressBackend"));
+        Self { log, imp: Arc::new(imp) }
+    }
+
+    pub(crate) async fn discover_peers(
+        &self,
+    ) -> Result<Vec<PeerAddress>, DiscoverPeersError> {
+        let log = self.log.new(slog::o!("task" => "discover_peers"));
+        slog::debug!(log, "discovering peers");
+
+        self.imp.discover_peers().await
+    }
+
+    pub(crate) async fn send_report_to_peer(
+        &self,
+        peer: PeerAddress,
+        update_id: Uuid,
+        report: EventReport,
+    ) -> Result<SendReportStatus, ClientError> {
+        let log = self.log.new(slog::o!("peer" => peer.to_string()));
+        // For each peer, report it to the network.
```

Comment:
Tiny nit - not sure this comment applies anymore? ("For each" sounds to me like we'll be looping over multiple peers, but that happens external to us now.)

Reply:
fixed, thanks!